rye/src/pyproject.rs

```diff
@@ -395,6 +395,21 @@ impl PyProject {
         None
     }
 
+    /// Set the target Python version.
+    pub fn set_target_python_version(&mut self, version: &PythonVersionRequest) {
+        let mut marker = format!(">= {}", version.major);
+        if let Some(minor) = version.minor {
+            marker.push('.');
+            marker.push_str(&minor.to_string());
+        }
+
+        let project = self
+            .doc
+            .entry("project")
+            .or_insert(Item::Table(Table::new()));
+        project["requires-python"] = Item::Value(Value::String(Formatted::new(marker)));
```

Comment:
Previously I was just accessing the doc directly like
```rust
self.doc["project"]["requires-python"] = ...
```
That didn't feel *safe*.